src/pyggdrasil/tree_inference/_mcmc.py

```diff
@@ -226,8 +228,10 @@ def _validate_move_probabilities(move_probabilities: MoveProbabilities, /) -> No
 def _mcmc_kernel(
     key: random.PRNGKeyArray,
     tree: Tree,
+    data: MutationMatrix,
+    theta: ErrorRates,
     move_probabilities: MoveProbabilities,
-    logprobability_fn: Callable[[Tree], float],
+    logprobability_fn: Callable[[MutationMatrix, Tree, ErrorRates], float],
```

Comment:
@pawel-czyz I realized that the implementation of the mcmc kernel, initially sketched out by you, feature a `log_probability` function that does not take the `data: Mutation matrix`nor  the `theta:ErrorRates`. Was this by design ? - Currently my `log_prob.logprobability_fn()` requires these arguments, hence these changes here. Is there a smarter way to pass these - as they don't change through the run - perhaps a rare case for a`global` variables.

Reply:
I think both options are fine. Essentially, I sketched it having a [curried](https://en.wikipedia.org/wiki/Currying) version of the function in mind:
you can have
```python

def logprob_all(data, tree, rates) -> float:
    return some_expression(...)
```
or:

```python

def create_logprob(data, rates) -> Callable:
    def logprob_(tree) -> float:
        return logprob_all(data, tree, rates)
    return logprob_
```

Currying is a common trick in functional programming and BlackJAX employs it often, but it doesn't mean that we have to (feel free to change the sketch :slightly_smiling_face: )